### PR TITLE
Apm 372025 make max log length configurable

### PR DESCRIPTION
**Pull request by lukasz-parulski-dynatrace**

### CODE REVIEW

File: dynatrace-aws-logs.sh [hunk 1]
```diff
@@ -78,13 +81,16 @@ arguments:
                         Enables checking SSL certificate of the target Active Gate. By default (if this option is not provided) certificates aren't validated
   --stack-name STACK_NAME
                         Optional. The name for the CloudFormation stack in which the resources will be deployed. This defaults to \"$DEFAULT_STACK_NAME\"
+  --max-log-length MAX_LOG_CONTENT_LENGTH
+                        Optional, defaults to 8192. Defines the max log length after which a log will be truncated.
+                        Requires a cluster side change for values over 8192 (Manage -> Settings -> Internal -> Indexed Log Storage Settings -> Event content maximum byte size).
```
Comment (mmajcher) [inline dynatrace-aws-logs.sh:86]: I am not sure "Settings \ Internal" is accessible for people. I think you have to contact Dynatrace ONE for this change. Please double check, ask logs team

File: tests/perf/logs/test_transformation_full.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 from util.context import Context
 
 BATCH_METADATA = BatchMetadata("444000444", "us-east-1", "aws")
-CONTEXT = Context("function-name", "dt-url", "dt-token", False, False, "log.forwarder")
+CONTEXT = Context("function-name", "dt-url", "dt-token", False, False, "log.forwarder", 8192)
```
Comment (dtwwolinski) [inline tests/perf/logs/test_transformation_full.py:26]: A magic number, Imho consider changing the 8192 into a constant, eg 
``` DYNATRACE_LOG_INGEST_CONTENT_MAX_LENGTH = 8192 ``` and import it for each usage of this value.

File: dynatrace-aws-logs.sh (hunk 1 above)
Comment (dtwwolinski) [reply dynatrace-aws-logs.sh]: @lukasz-parulski-dynatrace What about this one?

Comment (lukasz-parulski-dynatrace) [reply dynatrace-aws-logs.sh]: It's corrected already :D
image

**Review by dtwwolinski — DISMISSED** [review]

**Review by mmajcher — APPROVED** [review]

**Review by mmajcher — APPROVED** [review]

**Pull request merged**